server/app/services/export/QuestionJsonPresenter.java

```diff
@@ -0,0 +1,275 @@
+package services.export;
+
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.i18n.phonenumbers.NumberParseException;
+import com.google.i18n.phonenumbers.PhoneNumberUtil;
+import com.google.i18n.phonenumbers.Phonenumber;
+import com.typesafe.config.Config;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import javax.inject.Inject;
+import services.LocalizedStrings;
+import services.Path;
+import services.applicant.AnswerData;
+import services.applicant.question.CurrencyQuestion;
+import services.applicant.question.DateQuestion;
+import services.applicant.question.FileUploadQuestion;
+import services.applicant.question.MultiSelectQuestion;
+import services.applicant.question.NumberQuestion;
+import services.applicant.question.PhoneQuestion;
+import services.applicant.question.Question;
+import services.applicant.question.Scalar;
+import services.applicant.question.SingleSelectQuestion;
+import services.question.LocalizedQuestionOption;
+import services.question.types.QuestionType;
+
+public interface QuestionJsonPresenter<Q extends Question> {
```

Comment:
**suggestion (blocking):** add javadoc for this class explaining the overall pattern

Reply:
Done.